src/main/scala/webservice/CromIamApiService.scala

```diff
@@ -14,132 +19,179 @@ trait SwaggerService extends SwaggerUiResourceHttpService {
 
 trait CromIamApiService extends Directives with SprayJsonSupport with DefaultJsonProtocol with RouteConcatenation {
 
-  def returnInternalServerError(msg: String)  = HttpResponse(InternalServerError, entity = msg)
+  implicit val system: ActorSystem
+  implicit def executor: ExecutionContextExecutor
+  implicit val materializer: ActorMaterializer
 
-  val workflowRoutes = queryRoute ~ queryPostRoute ~ workflowOutputsRoute ~ submitRoute ~ submitBatchRoute ~
+  def cromwellInterface: String
+  def cromwellPort: Int
+
+  // Header that Akka HTTP adds to every request on receive.
+  // We get an warning in logs if we don't strip it out before sending the request to cromwell
+  // HTTP header ‘Timeout-Access: <function1>’ is not allowed in requests
+  val timeoutAccessHeader = "Timeout-Access"
+
+  private[webservice] def forwardToCromwell(httpRequest: HttpRequest): Future[HttpResponse] = {
+    val cromwellRequest = httpRequest
+      .copy(uri = httpRequest.uri.withAuthority(cromwellInterface, cromwellPort))
+      .withHeaders(httpRequest.headers.filterNot(header => header.name == timeoutAccessHeader))
+
+    Http().singleRequest(cromwellRequest)
+  }
+
+  val workflowRoutes: Route = queryRoute ~ queryPostRoute ~ workflowOutputsRoute ~ submitRoute ~ submitBatchRoute ~
     workflowLogsRoute ~ abortRoute ~ metadataRoute ~ timingRoute ~ statusRoute ~ backendRoute
 
-  val engineRoutes = statsRoute ~ versionRoute
+  val engineRoutes: Route = statsRoute ~ versionRoute
 
-  val allRoutes = workflowRoutes ~ engineRoutes
+  val allRoutes: Route = workflowRoutes ~ engineRoutes
 
-  def statusRoute =
+  def statusRoute: Route =
     path("api" / "workflows" / Segment / Segment / "status") { (version, possibleWorkflowId) =>
       get {
-        complete {
-          returnInternalServerError("workflow status")
+        extractRequest { req =>
+          complete {
+            forwardToCromwell(req)
+          }
         }
       }
     }
 
-  def queryRoute =
+  def queryRoute: Route =
     path("api" / "workflows" / Segment / "query") { version =>
       parameterSeq { parameters =>
         get {
-          complete {
-            returnInternalServerError("query get")
+          extractRequest { req =>
+            complete {
+              forwardToCromwell(req)
+            }
```

Comment:
Yeah, I thought about that... I'm gonna hold off on that until after we get the extra validation steps in, in case we end up with a premature abstraction that we just end up unpicking.